src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1179,20 +1179,17 @@ IsSelectClause(Node *clause)
 	Var *firstColumn = NULL;
 	Index firstColumnTableId = 0;
 	bool isSelectClause = true;
+	NodeTag nodeTag PG_USED_FOR_ASSERTS_ONLY = nodeTag(clause);
 
-	/* we currently consider the following nodes as select clauses */
-	NodeTag nodeTag = nodeTag(clause);
-	if (!(nodeTag == T_OpExpr || nodeTag == T_ScalarArrayOpExpr ||
-		  nodeTag == T_NullTest || nodeTag == T_BooleanTest))
-	{
-		return false;
-	}
+	/* planner already errored out for subqueries in WHERE clause */
+	Assert(nodeTag != T_SubLink);
```

Comment:
Hm. I think we should rather make this a runtime error check. It'll have weird consequences if it slips through, and the check is cheap. It's not unlikely to slip through imo, inlining and such can make things harder to detect.